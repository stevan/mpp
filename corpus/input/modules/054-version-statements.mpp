# Version with 'v' prefix
use v5.40;

# Version without 'v' prefix (also valid)
use 5.040;

# Version with feature bundle
use v5.38;

# Code after version statement
my $x = 10;
