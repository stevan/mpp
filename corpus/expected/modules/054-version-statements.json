[
  {
    "type": "Use",
    "version": "v5.40"
  },
  {
    "type": "Use",
    "version": "5.040"
  },
  {
    "type": "Use",
    "version": "v5.38"
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$x"
    },
    "initializer": {
      "type": "Number",
      "value": "10"
    }
  }
]